EmbeddedSocialClient/sdk/src/main/java/com/microsoft/embeddedsocial/server/model/notification/GetNotificationFeedResponse.java

```diff
@@ -17,21 +17,22 @@
 public class GetNotificationFeedResponse extends FeedUserResponse implements ListResponse<ActivityView> {
 
 	private List<ActivityView> activities;
+	// Setting this handle is required to properly update the "last-read" notification
 	private String deliveredActivityHandle;
 
 	public GetNotificationFeedResponse(List<ActivityView> activities) {
 		this.activities = activities;
-
-		// FIXME fix setting this handle
-		this.deliveredActivityHandle = "";
+		this.deliveredActivityHandle = activities.get(0).getHandle();
 	}
 
 	public GetNotificationFeedResponse (FeedResponseActivityView response) {
 		activities = new ArrayList<>();
 		for (com.microsoft.embeddedsocial.autorest.models.ActivityView view : response.getData()) {
 			activities.add(new ActivityView(view));
+			// Update the deliveredActivityHandle to be the most recent activity handle
+			this.deliveredActivityHandle = (activities.isEmpty() ? "" : activities.get(0).getHandle());
```

Comment:
Shouldn't this be following the for statement rather than being inside of it. If I understand correctly, we can set the deliveredActivityHandle once after executing the for-loop.

Reply:
Yes! I accidentally copied / pasted in the incorrect place from my modified code---thanks for catching it